alphastats/llm/uniprot_utils.py

```diff
@@ -431,23 +456,162 @@ def select_uniprot_id_from_feature(
+    result: Union[Dict, str],
     fields: list,
-):
+) -> Union[str, Dict]:
+    """
+    Wrapper around extraction of meaningful entries from uniprot json output. Handles errors and allows limiting the retrieved field keys.
+    Args:
+        result (dict): The UniProt result from which to extract information.
+        fields (list): A list of fields to extract from the result.
+    Returns:
+        dict or str: A dictionary containing the extracted field information, or the result itself if it is a string.
+    """
     if isinstance(result, str):
         return result
-    information = str({k: v for k, v in extract_data(result).items() if k in fields})
-    # TODO: Handle fields to format nice artifacts.
+    information = {
+        k: v
+        for k, v in _extract_annotations_from_uniprot_data(result).items()
+        if k in fields
+    }
     return information
 
 
-def get_information_for_feature(
+def _format_uniprot_field(
+    field: str, content: Union[str, List, Dict, None]
+) -> Union[str, None]:
+    """
+    Formats the content of a UniProt field based on the specified field type.
+    Args:
+        field (str): The type of UniProt field to format. Should be one of the ExtractedFields enum values.
+        content (Union[str, List, Dict, None]): The content to format. The type and structure of this parameter depend on the field type.
+    Returns:
+        Union[str, None]: A formatted string representation of the content for the specified field, or None if the content is None or empty.
+    """
+
+    if content is None:
+        return None
+    if field == ExtractedFields.NAME:
+        return (
+            None
+            if content["recommendedName"] is None
+            else " is called " + content["recommendedName"]
+            if "alternativeNames" not in content
+            else " is called "
+            + content["recommendedName"]
+            + " (or "
+            + "/".join(content["alternativeNames"])
+            + ")"
+        )
+    if field == ExtractedFields.GENE:
+        return (
+            " without a gene symbol"
+            if "geneName" not in content or content["geneName"] is None
+            else " " + content["geneName"]
+        )
+    if field in [
+        ExtractedFields.FUNCTIONCOMM,
+        ExtractedFields.SUBUNITCOMM,
+        ExtractedFields.TISSUE,
+    ]:
+        return " ".join(content) if len(content) > 0 else None
+    if field == ExtractedFields.INTERACTIONS:
+        return (
+            "Interacts with " + ", ".join([i["interactor"] for i in content]) + "."
+            if len(content) > 0
+            else None
+        )
+    if field == ExtractedFields.SUBCELL:
+        return "Locates to " + ", ".join(content) + "." if len(content) > 0 else None
+    if field == ExtractedFields.GOP:
+        return (
+            "The protein is part of the GO cell biological pathway(s) "
+            + ", ".join([el["name"] for el in content])
```

Comment:
please extract `", ".join([el["name"] for el in content])` to a variable

Reply:
and/or: unify these more:
```
if field == ExtractedFields.SUBCELL:
  prefix = "Locates to "
elif ...
```

and then have one statement in the end that does the formatting

or: use a `dict` `{ExtractedFields.SUBCELL: "Locates to " , ...}` to save on `ifs`

Reply:
Added TODO. Thank you for the suggestions. The big benefit of ifs is that is easily expandable without breaking anything.